src/kemal/middleware/filters.cr

```diff
@@ -55,22 +80,15 @@ module Kemal::Middleware
   end
 end
 
-def add_filters
-  unless filter = Kemal.config.handlers.any? { |handler| handler.is_a? Kemal::Middleware::Filter }
-    filter = Kemal::Middleware::Filter.new
-    Kemal.config.add_handler filter
-  end
-  filter
-end
-
-def before(path = "*", options = {} of Symbol => String, &block : HTTP::Server::Context -> _)
-  filter = (Kemal.config.handlers.find { |handler| handler.is_a? Kemal::Middleware::Filter } || add_filters) as Kemal::Middleware::Filter
-  raise Kemal::Middleware::Filter::BeforeFilterAlreadyDefinedException.new(path) if filter.filter_for_path_type_defined?(path, :before)
-  filter.add :before, path, options, &block
-end
+# All the helper methods available are:
+#  - before_all, before_get, before_post, before_put, before_patch, before_delete
+#  - after_all, after_get, after_post, after_put, after_patch, after_delete
 
-def after(path = "*", options = {} of Symbol => String, &block : HTTP::Server::Context -> _)
-  filter = (Kemal.config.handlers.find { |handler| handler.is_a? Kemal::Middleware::Filter } || add_filters) as Kemal::Middleware::Filter
-  raise Kemal::Middleware::Filter::AfterFilterAlreadyDefinedException.new(path) if filter.filter_for_path_type_defined?(path, :after)
-  filter.add :after, path, options, &block
-end
+ALL_METHODS = %w(get post put patch delete all)
+{% for type in ["before", "after"]%}
```

Comment:
I put it there first but then I thought since this is a middleware and optional it would make more sense to keep it separate from the `Kemal` dsl.
Are you sure :smile: ?